src/bytes.js

```diff
@@ -0,0 +1,1522 @@
+                }
+
+                encoding = encoding.$jsstr();
+                if (encoding === "ascii") {
+                    const data = [];
+                    for (let i in source) {
+                        const val = source[i].charCodeAt(0);
+
+                        if (val < 0 || val > 127) {
+                            if (errors === "strict") {
+                                const hexval = makehexform(val);
+                                throw new Sk.builtin.UnicodeEncodeError("'ascii' codec can't encode character '" + hexval + "' in position " + i + ": ordinal not in range(128)");
+                            } else if (errors === "replace") {
+                                data.push(63); // "?"
+                            }
+                        } else {
+                            data.push(val);
+                        }
+                    }
+                    ret = Uint8Array.from(data);
+                } else if (encoding === "utf-8") {
+                    ret = new TextEncoder().encode(source);
+                } else {
+                    throw new Sk.builtin.LookupError("unknown encoding: " + encoding.v);
+                }
+                view = new DataView(uint8ToBuffer(ret));
+            } else {
+                throw new Sk.builtin.TypeError("encoding without a string argument");
+            }
+        } else {
+            throw new Sk.builtin.TypeError("bytes() argument 2 must be str, not " + Sk.abstr.typeName(encoding));
+        }
+    }
+
+    this.v = view;
+    this.__class__ = Sk.builtin.bytes;
+
+    return this;
+};
+
+var makehexform = function (num) {
+    var leading;
+    if (num <= 265) {
+        leading = "\\x";
+    } else {
+        leading = "\\u";
+    }
+    num = num.toString(16);
+    if (num.length === 3) {
+        num = num.slice(1,3);
+    }
+    if (num.length  === 1) {
+        num = leading + "0" + num;
+    } else {
+        num = leading + num;
+    }
+    return num;
+};
+
+Sk.abstr.setUpInheritance("bytes", Sk.builtin.bytes, Sk.builtin.seqtype);
+
+Sk.builtin.bytes.prototype["$r"] = function () {
+    var ret;
+    var i;
+    var num;
+    ret = "";
+    for (i = 0; i < this.v.byteLength; i++) {
+        num = this.v.getUint8(i);
+        if ((num < 9) || (num > 10 && num < 13) || (num > 13 && num < 32) || (num > 126)) {
+            ret += makehexform(num);
+        } else if (num === 9 || num === 10 || num === 13 || num === 92) {
+            switch (num) {
+                case 9:
+                    ret += "\\t";
+                    break;
+                case 10:
+                    ret += "\\n";
+                    break;
+                case 13:
+                    ret += "\\r";
+                    break;
+                case 92:
+                    ret += "\\\\";
+                    break;
+            }
+        } else {
+            ret += String.fromCharCode(num);
+        }
+    }
+    ret = "b'" + ret + "'";
```

Comment:
I'm happy to address this one - I think i fixed it in a version I was working on.

Reply:
I "fixed" it, in the sense that there will not be an error now.  (It will do b'\''.)  If you have a fix that converts that to b"'", great.  I didn't think that was worth it, but I guess it's actually not that much more difficult.